### PR TITLE
V8 ai support

### PR DESCRIPTION
**Pull request by xxshubhamxx**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [ ] Back-ported PR at `#XXXXX`

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 1]
```diff
@@ -47,6 +47,7 @@ import type Percy from './Percy/Percy.js'
 import { sendStart, sendFinish } from './instrumentation/funnelInstrumentation.js'
 import BrowserStackConfig from './config.js'
 import { setupExitHandlers } from './exitHandler.js'
+import aiSDK from '@browserstack/ai-sdk-node'
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:50]: This package doesn't seem to exist yet, I think this should happen first before we merge this in.

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 2]
```diff
@@ -190,6 +191,31 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         } catch (err: unknown) {
             PercyLogger.error(`Error while setting best platform for Percy snapshot at worker start ${err}`)
         }
+        const setupTcgConfigFile = (tcgConfig: any) => {
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:194]: I see a lot of `any` typing here, we should get this properly typed.

File: packages/wdio-browserstack-service/src/service.ts [hunk 3]
```diff
@@ -110,6 +113,97 @@ export default class BrowserstackService implements Services.ServiceInstance {
...
+            //   logger.debug(`Cound not setup tcgAuth config file due to error: ${err}`)
+            }
+
+        }
+
+        //   const isUndefinedOrNull: any = val:any => {return (val === undefined || val === null || val === '')}
+
+        //   const isNotUndefinedOrNull: any = val:any => {return !isUndefinedOrNull(val)}
+
+        //   const isString: any = val:any => {return (isNotUndefinedOrNull(val) && typeof val === 'string')}
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/service.ts:140]: Are these still needed?

File: packages/wdio-browserstack-service/src/service.ts [hunk 4]
```diff
@@ -110,6 +113,97 @@ export default class BrowserstackService implements Services.ServiceInstance {
...
+                        console.log('Executing logData script' + script)
+                        await browser.execute(script)
+                    }
+                    if (!result.error) {
+                        return result
+                    }
+                    script = await aiSDK.BrowserstackHealing.healFailure(using1, value1, undefined, undefined, authInfo.userID, authInfo.groupID, sessionId, undefined, undefined, authInfo.groupAIEnabled, tcgDetails)
+                    if (script) {
+                        await browser.execute(script)
+                        console.log('Executing healing script' + script)
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/service.ts:190]: left over `console.log` statement

File: packages/wdio-browserstack-service/src/service.ts [hunk 5]
```diff
@@ -110,6 +113,97 @@ export default class BrowserstackService implements Services.ServiceInstance {
...
+                const sessionId: any = browser.sessionId
+                console.log('HEALING!!!!, patching find element')
+                const constants:any = {
+                    tcgRegion: 'use',
+                    tcgUrl: 'https://tcg.browserstack.com',
+                }
+                let tcgDetails = `{"region": "${constants.tcgRegion}", "tcgUrls": {"${constants.tcgRegion}": {"endpoint": "${constants.tcgUrl.split('s://')[1]}"}}}`
+                tcgDetails = isNotUndefinedOrNull(tcgDetails) && isString(tcgDetails) ? tcgDetails.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
+                const using1: string = isNotUndefinedOrNull(using) && isString(using) ? using.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
+                const value1: string = isNotUndefinedOrNull(value) && isString(value) ? value.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/service.ts:175]: Let's avoid such numbering for variables, if `using` or `value` is already used, have a more concrete name for this assignment.

**Review by christian-bromann — CHANGES_REQUESTED** [review]

**Comment by xxshubhamxx** [comment]
Hi @christian-bromann, thanks for inputs.
This is currently a work in progress. Converted this to a draft PR. Will update you when we are finished with the changes.

**Pull request closed without merging**